### ISSUE TITLE
Linux: Add Alsa Audio Support

### ISSUE THREAD

**Issue by mtkarp** [bluefireteam/audioplayers#1594]
### Checklist

- [X] I made sure that the issue I am raising doesn't already exist

### Use case / Problem

AudioPlayers currently interacts with PulseAudio. Constrained environments such as [Ubuntu Core](https://ubuntu.com/core) do not have PulseAudio available. Alsa is available on such platforms and providing support for Alsa will allow usage of the package in such environments.

### Proposal / Solution

No breaking changes are expected. Support of Alsa can be isolated from PulseAudio (and other platform code).

An API to configure which audio server to use will be necessary. This would preferably be done by the user at initialization of the AudioPlayer instance. The initialization can default to PulseAudio to minimize impact to existing users. It should not be necessary to switch between Alsa and PulseAudio dynamically - e.g., the user need not be able to play one sound on Alsa and the next on PulseAudio.

### Affected platforms

Linux

### Platform details

This feature request is isolated to the Linux Platform since Alsa only exists in the Linux environment.

### Related issues / more information

Reference the initial [Feature Request](https://github.com/bluefireteam/audioplayers/issues/798) on providing Linux Platform Support. A similar approach can be taken here with respect to interfacing with the underlying audio server. Note that a decision was made in this feature request to interface with PulseAudio.

### Working on PR

no way

**Comment by Gustl22** [comment]
We work with gstreamer, which uses pulseaudio as default. One could add a LinuxAudioContext and set the bits to enable ALSA:
https://gstreamer.freedesktop.org/documentation/tutorials/playback/digital-audio-pass-through.html?gi-language=c#inner-workings-of-gstreamer-audio-sinks

But according to the docs, gstreamer is already using ALSA if pulseaudio is not avavilable:
https://wiki.archlinux.org/title/PulseAudio/Examples#PulseAudio_as_a_minimal_unintrusive_dumb_pipe_to_ALSA

**Comment by Gustl22** [comment]
@mtkarp did you try to use the lib, when PulseAudio was not available? What is the output/error on such systems?

**Comment by mtkarp** [comment]
@Gustl22 Sorry for not responding earlier. We have solved this issue while still using the AudioPlayers library. I recommend we close this feature request as there are two workarounds for anyone facing the same issue.

1. As you mentioned, `gstreamer` can be used in combination with ALSA to allow AudioPlayers to operate in a constrained environment. `gstreamer` (and possibly other libraries) needs to be added to the environment (in this case, snap) the developer is creating.
2. If the application needs higher level audio support, `PulseAudio` can also be installed by creating a snap (needed for the Ubuntu Core environment) with the `PulseAudio` binary.

We tested both of these approaches and the AudioPlayers library works as expected in both scenarios.
